### DISCUSSION TITLE
Request:サイト内のリンクを自動リファクタリング出来るようにしてほしい

### DISCUSSION THREAD

**Discussion by marioninc** [growilabs/growi#7734] [Ideas / Requests] [answered]
ページを移動/名前変更したときに
サイト内のリンクを自動リファクタリング出来るようにしてほしい

ページの名称を変更
もしくは、プロジェクトの終了などによりフォルダにまとめるときに、そのページに対するリンクが切れてしまいます。
ページの移動や、名前変更の時に被リンクを追ってリファクタリングを出来るようになれば便利だなと思いました。

**Comment by takahyon** [comment]
+1

**Comment by yuki-takei** [comment]
@marioninc 利用方法の議論かなと思います。現状はページ内のリンクをどのように書いていらっしゃるのでしょうか？

v5 からはページパス形式の URL (e.g. https://example.com/Sandbox) でアクセスしても最終的に permalink にリダイレクトされるようになったので、自然とページ内リンクも permalink で記述する人が増えるかなという想定をしていました。その場合、ページ名称を変更してもリンク先はそのままになるかと思います。

一方でページパス形式での記述を禁止するものではないので、ご要望いただいたような被リンクの検出は追加で検討する機能になりますね。

**Comment by marioninc** [comment]
@yuki-takei 回答ありがとうございます。
permalinkでリンクをすることで、ページ名を変更（移動）してもリンク先が維持されるのは知っていました。
利用方法として、permalinkを使うことが基本なのであればそれでも良いと思います。
permalinkを使うのか、ページパス形式を使うのかは、おっしゃる通り利用方法の話ですね。

> permalink にリダイレクトされるようになったので、自然とページ内リンクも permalink で記述する人が増えるかなという想定をしていました。

については、タイトル横のクリップボートにコピーの機能で、ページURLの方がパーマリンクより上位にあることがページパス形式を使われてしまう一端にあるかと思います。

場合によってWiKiを使うのが、技術者だけではない場合（管理者より使用方法を説教出来ない場合）があると思うので、
そのようなユーザーを含む場合、ページパス形式の方が理解させやすいと思っています。
また、ページパス形式を禁止できない以上使われてしまうというのを管理上どうやってリンク切れを防止していくのかがやはり課題になってしまうのかなと。
ならばリンクが自動的に維持される機能があったほうがいいなという提案でした。

**Comment by yuki-takei** [comment]
開発者会議で議論は行いました。一旦優先度Dに入れて仕様の継続議論となります。

[/GROWI村議議事録/20230203](https://dev.growi.org/63dc7604e1838eff0bc30bfd)

**Comment by yuki-takei** [comment, marked as answer]
ご要望をいただきありがとうございます。検討の結果、以下の理由から現時点での対応は難しいと判断しました。

- **permalink（共有リンク）を使用している場合**: ページ移動後も自動的にリダイレクトされるため、リファクタリングは不要です
- **ページパス（例: `/parent/child`）でリンクしている場合**: ページ移動によるリンク切れは現時点では仕様上の制限となります

なお、リンク切れの検出という観点では、[バックリンク表示機能（#11140）](https://github.com/growilabs/growi/discussions/11140) で将来的に対応できる可能性があります。

引き続き GROWI をよろしくお願いします。
